### ISSUE TITLE
ResolvePackageAssets task should log package dependency chain on each item

### ISSUE THREAD

**Issue by kartheekp-ms** [dotnet/sdk#17207]
https://github.com/NuGet/Home/issues/10796#issue-865359017

The `ResolvePackageAssets` task takes in PackageReference items and produces `RuntimeCopyLocalItems`. It is not clear why a certain file ends up in RuntimeCopyLocalItems.

A new metadata should be logged on each file that shows the dependency chain from a root package to the package that caused this file inclusion.

This information is available in project.assets.json but is very hard to parse.

Having this metadata would significantly simplify investigations answering questions such as "why is this file being copied to output" and "which NuGet package was the root package that eventually caused this file to be included".

Mockup:
![image](https://user-images.githubusercontent.com/679326/115771623-08d1f780-a363-11eb-8795-19818e36d800.png)

**Comment by kartheekp-ms** [comment]
Need to think about the metadata name, separator (maybe ` -> `?) etc. But the idea is I need to see which package dependency chain led to each resolved asset.

_Originally posted by @KirillOsenkov in https://github.com/NuGet/Home/issues/10796#issuecomment-825113054_

**Comment by KirillOsenkov** [comment]
@clairernovotny @jeffkl @nkolev92 I really miss this functionality.

It is often very important to know which direct PackageReference caused (directly or indirectly) a given NuGet package dependency. Sometimes a file from a random package is being copied to output and you need to figure out which "root" package in the project (or perhaps an inherited one from referenced projects??) has caused this NuGet package to bring in assets.

I suppose I could reverse-engineer this from project.assets.json, but this file isn't always available.

**Comment by KirillOsenkov** [comment]
@dsplaisted @rainersigwald

**Comment by dsplaisted** [comment]
This doesn't seem to me like something that we should be doing in ResolvePackageAssets.  We'd need to reconstruct the package graph and walk it, which I don't think we currently do at all.  Also, there's not necessarily just one path of dependencies, and there may be more than one root PackageReference bringing something in.

I think if you search for the asset in the Solution Explorer in Visual Studio, it may show you the dependency chain.  Barring that, I think the best solution would be to have a tool that can process the assets file and produce this information.
